src/routes/load/components/OwnerList/index.jsx

```diff
@@ -0,0 +1,177 @@
+// @flow
+import * as React from 'react'
+import Block from '~/components/layout/Block'
+import { withStyles } from '@material-ui/core/styles'
+import Field from '~/components/forms/Field'
+import { required } from '~/components/forms/validator'
+import TextField from '~/components/forms/TextField'
+import OpenInNew from '@material-ui/icons/OpenInNew'
+import Identicon from '~/components/Identicon'
+import OpenPaper from '~/components/Stepper/OpenPaper'
+import Col from '~/components/layout/Col'
+import Row from '~/components/layout/Row'
+import Link from '~/components/layout/Link'
+import Paragraph from '~/components/layout/Paragraph'
+import Hairline from '~/components/layout/Hairline'
+import {
+  xs, sm, md, lg, border, secondary,
+} from '~/theme/variables'
+import { getOwnerNameBy } from '~/routes/open/components/fields'
+import { getEtherScanLink, getWeb3 } from '~/logic/wallets/getWeb3'
+import { FIELD_LOAD_ADDRESS } from '~/routes/load/components/fields'
+import { getGnosisSafeContract } from '~/logic/contracts/safeContracts'
+
+
+const openIconStyle = {
+  height: '16px',
+  color: secondary,
+}
+
+const styles = () => ({
+  details: {
+    padding: lg,
+    borderRight: `solid 1px ${border}`,
+    height: '100%',
+  },
+  owners: {
+    display: 'flex',
+    justifyContent: 'flex-start',
+  },
+  ownerNames: {
+    maxWidth: '400px',
+  },
+  ownerAddresses: {
+    alignItems: 'center',
+    marginLeft: `${sm}`,
+  },
+  address: {
+    paddingLeft: '6px',
+  },
+  open: {
+    paddingLeft: sm,
+    width: 'auto',
+    '&:hover': {
+      cursor: 'pointer',
+    },
+  },
+  title: {
+    padding: `${md} ${lg}`,
+  },
+  owner: {
+    padding: `0 ${lg}`,
+    marginBottom: '12px',
+  },
+  header: {
+    padding: `${sm} ${lg}`,
+  },
+  name: {
+    marginRight: `${sm}`,
+  },
+})
+
+type LayoutProps = {
+  network: string,
+}
+
+type Props = LayoutProps & {
+  values: Object,
+  classes: Object,
+}
+
+type State = {
+  owners: Array,
+}
+
+class OwnerListComponent extends React.PureComponent<Props, State> {
+  state = {
+    owners: [],
+  }
+
+  mounted = false
+
+  componentDidMount = async () => {
+    this.mounted = true
+
+    const { values } = this.props
+    const safeAddress = values[FIELD_LOAD_ADDRESS]
+    const web3 = getWeb3()
+
+    const GnosisSafe = getGnosisSafeContract(web3)
+    const gnosisSafe = await GnosisSafe.at(safeAddress)
+    const owners = await gnosisSafe.getOwners()
+
+    if (!owners) {
+      return
+    }
+
+    if (this.mounted) {
+      this.setState(() => ({ owners: owners.sort() }))
```

Comment:
I'd say this should be a redux action, because we'll need owners info in other components like transaction list, when sending funds, etc

Reply:
IIRC we already have owners info in redux store, so you can just pass it via props